calculator/serializers.py

```diff
@@ -54,4 +54,11 @@ class BalanceSerializer(serializers.Serializer):
     def to_representation(self, obj):
         if not obj["date"]:
             obj["date"] = datetime.now().astimezone(tz=timezone.utc)
-        return {"balance": Loan.objects.get(pk=obj["loan_id"]).get_balance(obj["date"])}
+        return {"balance": self.loan.get_balance(obj["date"])}
+    
+    def validate(self, data):
```

Comment:
@julianeventeu I think that without wanting to end up doing the same thing .. I had already sunk this in my pr

https://github.com/cesarbruschetta/api-fintech/blob/2259c76a6efa3dcd9b557ce81471dd014c277a0f/calculator/serializers.py#L41-L44

Reply:
Actually, they are different validations. Your validation doesn't allow inserting a payment with a date previous the date initial loan. My validation doesn't allow request a balance with a date previous the date initial loan. They are two different requests.

Reply:
ok